luaui/Widgets/cmd_nano_range_check.lua

```diff
@@ -0,0 +1,277 @@
+-- Default is 20. But modern hardware could handle more. 20 is sufficient for any game case.
+local COMMAND_LIMIT = 20
+-- variables
+local is_play = false
+local counter = 0
+local flag_listening = false
+local flag_processed = true
+local current_towers = {}
+local command_budget = 0
+local to_be_cleared = {} -- stack of uIDs to be cleared
+local new_orders = {} -- stack of {uid = {cmdArr},...} to be flag_processed
+local radius_cache = LRUCache:new(10) -- LRU cache to reduce the number of calls for model radius
+-- !SECTION Settings and other variables
+
+-- SECTION Helper functions
+-- converts a table to a Set()-like
+local function tableToSet(t) -- coverts a table to a Set()-like
+    for _, key in ipairs(t) do t[key] = true end
+end
+
+-- Returns a float from 2 coordinates
+-- NOTE equal to Spring.GetUnitSeparation(u1, u2, false, false) Uses 3D distance not 2D
+-- This is faster then synch reading from the engine; jumping.
+-- 1.4e changed to use 2D distance x and z instead of 3D, as it seems that nanos work in 2D cylinders.
+local function calculateDistance2D(x1, y1, x2, y2)
+    return math.sqrt((x1 - x2)^2 + (y1 - y2)^2)
+end
+
+-- LRU caching of model radius, so we don't have to get it every time
+local function getUnitRadius(uID)
+    local radius = radius_cache:get(uID)
+    if not radius then
+        radius = Spring.GetUnitDefDimensions(Spring.GetUnitDefID(uID)).radius
+        radius_cache:put(uID, radius)
+    end
+    return radius
+end
+
+-- Checks if a command is valid
+local function isValidCommandID(commandID)
+    return commandID == CMD.REPAIR or commandID == CMD.GUARD or commandID == CMD.RECLAIM or commandID == CMD.ATTACK
+end
+-- !SECTION Helper functions
+
+-- SECTION Main functions
+local function checkTurretRange(uID)
+    local x, y, z = Spring.GetUnitPosition(uID)
+    local build_distance = Spring.GetUnitEffectiveBuildRange(uID, nil) -- Same as UnitDef.buildDistance; Not ambiguous
+    local is_changed = false
+    local is_first_cmd = true
+    local cmds = Spring.GetUnitCommands(uID, -1)
+    local new_cmds = {}
+    for i = #cmds, 1, -1  do
+        local cmd = cmds[i]
+        if isValidCommandID(cmd.id) then
+            local tuID = cmd.params[1]
+            local tx, ty, tz = Spring.GetUnitPosition(tuID)
+            if tx == nil then break end
+            local distance = calculateDistance2D(x, z, tx, tz)
+            local radius = getUnitRadius(tuID)
+            if distance < build_distance + radius then -- BP uses build_distance + radius (sphereical shape of model)
+                cmd.options.shift = not is_first_cmd
+                is_first_cmd = false
+                table.insert(new_cmds, {cmd.id, cmd.params, cmd.options}) -- building cmdArr
+            else
+                is_changed = true
+            end
+        end
+    end
+    if is_changed then
+        if #new_cmds > 0 then
+            table.insert(new_orders, {uID, new_cmds}) -- schedule for new order
```

Comment:
```suggestion
            table.insert(new_orders, uID, new_cmds) -- schedule for new order
```
Should this be inserting an entry with a key of `uID`, whose value is the table of `new_cmds`?